/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax="proto3";

package message;

// Envelope will encapsulate all messages
message Envelope {
    int32 messageId = 1;
    bytes message = 2;
}

// Knock Request is the usual check in
// msgId = 15
message Knock {
    string botId = 1;
    uint32 knockId = 2;
}

// Server always responds with task, but it may be empty
// in the demo, it will be simple -- launch cmd command
// msgId = 16
message Task {
    uint32 taskId = 1;
    string task = 2;
}

// Task Response is recieved instead of knock
message TaskResponse {
    string botId = 1;
    uint32 knockId = 2;
    uint32 taskId = 3;
    string task = 4;
}